Creating a honeypot for network security (#6197)
section-engineering-education/engineering-education: content/articles/creating-a-honeypot-for-network-security/index.md

- [Types of honeypots](#types-of-honeypots)
- [How honeypots operate](#how-honeypots-operate)
- [Merits and demerits](#merits-and-demerits)
- [Honeypot traps and web scraping](#honeypot-traps-and-web-scraping)
- [Avoiding honeypot traps](#avoiding-honeypot-traps)
- [Creating a honeypot](#creating-a-honeypot)
- [Honeypot software](#honeypot-software)
- [Conclusion](#conclusion)

### Types of honeypots
Honeypots are classified into the following:

1. Research honeypots
2. Production honeypots

They are described below.

- `Research honeypots:` Researchers use research honeypots to learn more about the black hat group's methods and goals. These honeypots are used to analyze corporate threats and learn how to mitigate them. Academic honeypots are used by research, military, and government groups to collect large amounts of data.

- `Production honeypots:` Production honeypots are famous for their low data collection and ease of usage. Adding production honeypots improves network security. Due to their little participation, production honeypots are easy to set up. Research honeypots provide more about the assaults and attackers.

Design-wise, honeypots are classed as follows:
 

> - Low-interaction honeypots
> - pure honeypots
> - high-interaction honeypots

Reviewer: ```suggestion
- Low-interaction honeypots
- Pure honeypots
- High-interaction honeypots
```

> `Low-interaction honeypots`- replicate attacker-requested services. Multiple virtual machines may be readily hosted on one physical system, and less code is needed, decreasing the virtual system's security complexity. Honey

Reviewer: Honey....is this paragraph unfinished?

`Pure honeypots`- are full-scale factories. The honeypot's network connection is bugged to monitor the attacker's activities. A separate piece of software isn't needed. Both pure honeypots and honeypot-like systems have benefits and downsides.

> High-interaction honeypots enable an attacker to waste time by mimicking production system activities. Using virtual PCs, a single server may host several honeypots. A quick honeypot penetration may restore it. While high-interaction honeypots are safer, they need more maintenance. It may be too pricey without virtual computers.

Reviewer: ```suggestion
`High-interaction honeypots` enable an attacker to waste time by mimicking production system activities. Using virtual PCs, a single server may host several honeypots. A quick honeypot penetration may restore it. While high-interaction honeypots are safer, they need more maintenance. It may be too pricey without virtual computers.
```

### How honeypots operate

> Most honeypots are found in the DMZ. The DMZ has web and email servers connected to the Internet. A firewall divides the corporate network from the DMZ. It protects an internal network from the outer world. It permits traffic to flow in key network areas.

Reviewer: What's a DMZ? Let it be clear to our readers

The invaders think these fake servers are legitimate, but they are honeypots. So they follow well-established protocols and base their behaviors on real-world acts. Occasionally, fake data is employed to improve the reliability of a piece of data. A honeypot may contain fraudulent credit card information or files that seem to contain such data. The honeypot duplicates all routine access control procedures and protocols to protect the data.

Attackers see honeypots in the DMZ as a susceptible location. And the more convincing they must be. How long should a hacker stay in a honeypot? An attacker might want to remain in the honeypot and collect as much information about the target firm as possible to avoid detection.

> ![The working of honeypots](/engineering-education/creating-a-honeypot-for-network-security/image2.png)

Reviewer: Is this your image? If not, have you added the sources?

### Merits and demerits
Let's look at the merits and demerits of using honeypots.

#### Advantages

> Experts say honeypots function when other IDS fail.

Reviewer: What's an IDS? Explain it to our readers

- Never utilize system admin accounts.
- Use a different email address.
- Rather than keeping sensitive data on the server, use bogus data.
- Only the physical server should be accessed.

In the case of a security compromise, virtual machines may be quickly turned down and re-built. Any virtual server or set of virtual machines requires a hypervisor.

**step 2** Set up the firewall and logs.

Now it's time to pick what honeypot will monitor. File modifications, Login attempts, and other actions. The administration must provide more logging since hackers may modify log files. As a precaution, save the honeypot's log files someplace unlikely to be hacked, such as the Windows logging tool or a cloud service.

The honeypot is in the DMZ, outside the firewall. The external firewall should only allow vital ports and block all others. These devices are ideal for diverting traffic from internal networks to public networks.

**step 3** Set up the honeypot
Unlike the internal firewall, the honeypot is a security concern. Offensive actors may access many ports. Don't open all ports. Bypassing a crucial server requires leaving or exploiting the honeypot.

> 4th Step: Testing Nmap and other port scanners may be used to examine a honeypot's settings. Assailants often do their port scans to find network weaknesses. An attacker may see what a hacker sees.

Reviewer: ```suggestion
**Step 2:** Set up the firewall and logs.

Now it's time to pick what honeypot will monitor. File modifications, Login attempts, and other actions. The administration must provide more logging since hackers may modify log files. As a precaution, save the honeypot's log files someplace unlikely to be hacked, such as the Windows logging tool or a cloud service.

The honeypot is in the DMZ, outside the firewall. The external firewall should only allow vital ports and block all others. These devices are ideal for diverting traffic from internal networks to public networks.

**Step 3:** Set up the honeypot
Unlike the internal firewall, the honeypot is a security concern. Offensive actors may access many ports. Don't open all ports. Bypassing a crucial server requires leaving or exploiting the honeypot.

**Step 4**: Testing Nmap and other port scanners may be used to examine a honeypot's settings. Assailants often do their port scans to find network weaknesses. An attacker may see what a hacker sees.
```

![An Nmap scan's output](/engineering-education/creating-a-honeypot-for-network-security/image1.png)

Nmap's port scanning capability can detect open and closed ports. Having a honeypot's ports open and closed increases authenticity.

Finally, administrators should set up a honeypot and watch server logs. An IDS may refuse access to unprotected ports to advise an attacker to search elsewhere.

> These issues must be addressed before honeypot manufacture. That the honeypot may attract attackers and capture data is critical. After the initial test, inspect and fine-tune the honeypot.

Reviewer: This sentence, "That the honeypot may attract attackers and capture data is critical." is not clear

### Honeypot software
Let's look at some of the Honeypot software:

- The Honeynet Project's Ghost USB. This free honeypot looks like a USB gadget. This device attracts and identifies viruses. Ghost USB offers both a command-line and a GUI.

- Gianluca Brindisi's Wordpot. In this software, the users may detect files used by hackers to obtain data from WordPress sites. WordPress plugins may duplicate a WordPress issue. It's GUI (CLI).

> - Attivo Inc. produces ThreatDefend. A self-learning honeypot/deception device. These artificial visible local area networks are profiled and modified (VLAN). Finding hacker tactics to breach the company's firewall is possible. This business specializes in APTs, which overcome standard security mechanisms.

Reviewer: Please add external links to these softwares

### Conclusion

> Honeypot traps have been used to catch cybercriminals for years. They may, however, hinder web scraping. Don't utilize honeypot traps to scrape publicly available data like price tracking or market research.

Reviewer: Honeypot traps have been used to catch cyber criminals for years. However, they may hinder web scraping. Don't utilize honeypot traps to scrape publicly available data like price tracking or market research.
